test/e2e/tests/test_references.py

```diff
@@ -0,0 +1,163 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Integration tests for resource references"""
+
+import json
+import time
+
+import pytest
+
+from acktest.k8s import condition
+from acktest.k8s import resource as k8s
+from acktest.resources import random_suffix_name
+from e2e import service_marker, CRD_GROUP, CRD_VERSION, load_resource
+from e2e.common.types import POLICY_RESOURCE_PLURAL, ROLE_RESOURCE_PLURAL
+from e2e.replacement_values import REPLACEMENT_VALUES
+from e2e import role
+from e2e import policy
+from e2e import tag
+
+DELETE_ROLE_TIMEOUT_SECONDS = 10
+# Little longer to delete the policy since it's referred-to from the role...
+DELETE_POLICY_TIMEOUT_SECONDS = 30
+CHECK_WAIT_AFTER_SECONDS = 10
+CHECK_WAIT_AFTER_REF_RESOLVE_SECONDS = 10
+MODIFY_WAIT_AFTER_SECONDS = 20
+MAX_SESS_DURATION = 3600 # Note: minimum of 3600 seconds...
+
+@pytest.fixture(scope="module")
+def referred_policy_name():
+    return random_suffix_name("referred-policy", 24)
+
+
+@pytest.fixture(scope="module")
+def referring_role(referred_policy_name):
+    role_name = random_suffix_name("referring-role", 24)
+    role_desc = "a role that refers to a policy"
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements['ROLE_NAME'] = role_name
+    replacements['ROLE_DESCRIPTION'] = role_desc
```

Comment:
lol Do we really need to have this as a parameter? Surely we could have hardcoded this into the YAML

Reply:
removed unnecessary params.